### ISSUE TITLE
Clarification: Clean Up & Reset Options and Deleting Fields or Pods 'staying' around

### ISSUE THREAD

**Issue by jimtrue** [pods-framework/pods#4235]
At present, under "Clean Up & Reset" in Pods Admin, Settings, we have these two options:

> ## Reset Pods
> 
> This will reset Pods settings, removes all of it's data, and performs a fresh install.
> 
> *Please Note: This does not remove any items from any Post Types, Taxonomies, Media, Users, or Comments data you have added/modified.*
> 
> ## Deactivate and Delete Pods Data
> 
> This will delete Pods settings, data, and deactivate itself once done. Your database will be as if Pods never existed.
> 
> *Please Note: This does not remove any items from any Post Types, Taxonomies, Media, Users, or Comments data you have added/modified.*
> 

For clarification, since the two notes both say _exactly_ the same thing: This doesn't sound like it actually deletes any Custom Post Types, Fields, etc. that are created. This sounds like it only clears out the Pods Configuration. And I just confirmed, this is actually the case.

- Post Types I create with Pods are still there in `wp_posts`
- Field Meta added and created with Pods is still there in `wp_postmeta`, including relationship fields pointing at the other post id.
- With Deactivate/Delete, it actually removes the wp_podsrel (although reactivating the plugin does bring it back, but it's empty at this point, even though unconfigured Pods created by Pods are still in posts/postmeta with their relationships).

I think we need a separate/better option (and more clarity on the above two descriptions or breaking them into their own section) specifically to delete the Pods and the fields you create with it. If you're deleting an extended User pod, it should delete the user_meta created with it, likewise with any extended object. Basically anything "we" Create, "we" should take responsibility for properly deleting from posts/meta/termmeta/etc. We know the configuration of those Pods/Fields _prior_ to Reset or Deactivate/Delete. Once we delete _our_ configuration controlling hidden fields, we no longer know what we created and what we didn't. That's a problem.

We obviously need to be very clear in these descriptions and give people a change to back out or type 'DELETE' (make sure you did a backup, etc.), but we need to be good about 'garbage collection'. Things we create and that are owned by us, should be able to be cleared up by us.

And if we keep the stuff out there, then we need to have a way to re-identify/scan/re-enable into Pods without going through wp_posts and wp_postmeta manually and re-adding the Pod and it's fields.

Thoughts?

**Comment by jimtrue** [comment]
Possible Feature Enhancement:
Import Pod from existing wp_post/wp_postmeta. It'd be connected by post_id, so we reasonably should be able to identify everything but what 'kind' of field it is. This may be the reason we kept the old posts/meta around, but that's not helpful unless you have an easy way to reconfigure them IN pods.

This would also open up the ability of re-creating ACF/Types/hand-registered MetaBox fields as extensions of their post types, etc.

**Comment by quasel** [comment]
I agree the 2 Options are almost the same I would  just leave reset 
( or if you insist ask after reset if pods should deactivate itself  but then there comes the next question shall we delete pods.. ) But I would add Reset Configuration and Delete all Data ;) 

![admin](https://cloud.githubusercontent.com/assets/9464217/26266494/df61714c-3ce5-11e7-9305-20705acf75bb.png)
If we delete data it should be feeling like a hell of confirmation ralley as we need to think about the worst user possible. No idea how far the current option goes  

Regarding restoring configuration: This is only an issue if the user "resets data" 
Correct me if I'm wrong but if I create a field and give it the correct "name" = meta_key  it just works again (there might be issues if I choose a different type but ;) ) 

Makes that sense?

**Comment by jimtrue** [comment]
This is an excellent way to explain what UnInstall or Reset does (from Beaver Builder). I also love the Setting Page itself with the vertical tab interface (for #2197 review)

![screenshot 2017-06-14 12 02 11](https://user-images.githubusercontent.com/3052875/27142659-afc06236-50f9-11e7-9270-9125f257955a.png)

**Comment by jimtrue** [comment]
We could keep all this on the Clear Cache area but more explicitly explain what it does and add the 'debug details' into a read-only text window with a quick 'copy to clipboard' button to make it easier for folks to provide Debug Details.

**Comment by pglewis** [comment]
Another one I'm pulling out of the Maintenance Releases project for now, in an effort to get the incoming/to do lists in better shape.

**Comment by jimtrue** [comment]
Adding Reference to #4369 as well.

**Comment by quasel** [comment]
Another one #3029
> Add actions links list to the Edit Pod screen, add filter to add additional actions from plugins, Move Delete All Items action into this screen and out of Manage Pods list submitted via Slack by sc0ttkclark

**Comment by sc0ttkclark** [comment]
The commit https://github.com/pods-framework/pods/commit/f10fa04e3fd6de4a38de9e8f184325915e2bccbe helps the text here, but we probably need another reset/delete option that will go through and reset all of the custom pods, and delete all of the custom fields for any extended pods too.
